### PR TITLE
Convert scxa h5ad into cxg

### LINKED ISSUE #1: Write scxa->cxg h5ad converter - obs only
CxG Standard - latest approved version: https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/4.0.0/schema.md

Example h5ad file on scxa FTP: https://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/sc_experiments/E-CURD-2/

Attempt at mapping obs between schemas here: https://github.com/VirtualFlyBrain/scxa-to-h5ad/issues/1#issue-1881521372

It is also worth checking vars for compliance (making sure gene name. ID and type are stored in a schema compliant way.

Finally - `obsm` stored embeddings.  Ideally this would be compliant too.

### PR DESCRIPTION
**Pull request by anitacaron**
Fixes #1

### CODE REVIEW

File: src/scxa2cxg.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        "inferred_cell_type_-_ontology_labels_ontology": "cell_type_ontology_term_id",
+        "authors_cell_type_-_ontology_labels": "cell_type",
+        "authors_cell_type_-_ontology_labels_ontology": "cell_type_ontology_term_id",
+        "organism_ontology": "organism_ontology_term_id",
+        "organism_part": "tissue",
+        "organism_part_ontology": "tissue_ontology_term_id",
+        "disease_ontology": "disease_ontology_term_id"
+        },
+        inplace=True
+    )
```
Comment (ubyndr) [inline src/scxa2cxg.py:65]: Do we need to validate their h5ad files to ensure that the necessary columns exist before attempting to rename them?

File: src/scxa2cxg.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    Change column names to be conform to CxG schema v4.0.0
+    """
+    ann_data = ad.read_h5ad(
+        f"downloads/{study}/{study}{H5AD_EXT_FILE}",
+        backed="r+"
+    )
+    new_obs = pd.DataFrame(
+        data=ann_data.obs.copy(),
+        index=ann_data.obs.index.copy()
+    )
```
Comment (ubyndr) [inline src/scxa2cxg.py:45]: `new_obs = ann_data.obs.copy()` should be enough, shouldn't it?

File: src/scxa2cxg.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        },
+        inplace=True
+    )
+
+    ann_data.obs = new_obs
+    ad.AnnData.write_h5ad(
+        ann_data,
+        f"downloads/{study}/{study}_modified{H5AD_EXT_FILE}",
+        compression="gzip"
+    )
```
Comment (ubyndr) [inline src/scxa2cxg.py:72]: I would have used `ann_data.write_h5ad(f"downloads/{study}/{study}_modified{H5AD_EXT_FILE}", compression="gzip")` instead

**Review by ubyndr — CHANGES_REQUESTED** [review]

File: src/scxa2cxg.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        "developmental_stage": "development_stage",
+        "developmental_stage_ontology": "developmental_stage_ontology_term_id",
+        "inferred_sex": "sex",
+        "inferred_sex_ontology": "sex_ontology_term_id",
+        "sex_ontology": "sex_ontology_term_id",
+        "cell_type_ontology": "cell_type_ontology_term_id", 
+        "inferred_cell_type_-_ontology_labels": "cell_type",
+        "inferred_cell_type_-_ontology_labels_ontology": "cell_type_ontology_term_id",
+        "authors_cell_type_-_ontology_labels": "cell_type",
+        "authors_cell_type_-_ontology_labels_ontology": "cell_type_ontology_term_id",
```
Comment (ubyndr) [inline src/scxa2cxg.py:58]: `cell_type` and `cell_type_ontology_term_id` are duplicated. Is this correct?

Comment (anitacaron) [reply src/scxa2cxg.py:58]: I found in some files they use different names for the cell type. I've got the obs columns for each experiment in here https://github.com/EBISPOT/scxa_kg/blob/3b76ab7c605c6fda263d00421f99494b16a81007/obs_columns_by_experiment.json

File: src/scxa2cxg.py (hunk 1 above)
Comment (anitacaron) [reply src/scxa2cxg.py:65]: It renames the one available in the file. It doesn't get an error if the column doesn't exist.

File: src/scxa2cxg.py (hunk 2 above)
Comment (anitacaron) [reply src/scxa2cxg.py:45]: thanks, I changed that

File: src/scxa2cxg.py (hunk 3 above)
Comment (anitacaron) [reply src/scxa2cxg.py:72]: ok, changed that.

**Review by ubyndr — APPROVED** [review]

**Comment by dosumis** [comment]
Wow - that was fast!

**Pull request merged**